src/main/scala/com/thirdship/libunit/units/MetricTSUnit.scala

```diff
@@ -0,0 +1,52 @@
+package com.thirdship.libunit.units
+
+import com.thirdship.libunit.utils.{ExactString, FuzzyString, WordString}
+import com.thirdship.libunit.utils.Helpers._
+import com.thirdship.libunit._
+
+class MetricTSUnit(unit: String, parseList: List[FuzzyString]) {
```

Comment:
```
private val baseUnit = "s"
private val compressedParseMap = Map(
    baseUnit.i  ->  List("sec".i, "second".w),
    "min".i     -> List("minute".w),
    "h".i       -> List("hr".w, "hour".w),
    "d".i       -> List("day".w),
    "y".i       -> List("yr".w, "year".w),
    "jiffy".i   -> List()
)

private val edges = List(
    new ScalarConversionEdge(baseUnit,  baseUnit,   1),
    new ScalarConversionEdge("min",     baseUnit,   60),
    new ScalarConversionEdge("h",       "min",      60),
    new ScalarConversionEdge("d",       "h",        24),
    new ScalarConversionEdge("y",       "d",        365.2422,   1),
    new ScalarConversionEdge(baseUnit,  "jiffy",    3e-24,      1)
) 

val data: AStarConvertibleTSUnitData = new AStarConvertibleTSUnitData(baseUnit, "Time", compressedParseMap, edges).createMetricUnits(baseUnit)
```

then, the . createMetricUnits would return a new AStarConvertibleTSUnitData where  the compressedParseMap and the edge list have been update in the same way. Note that the units are made for an arbitrary point, and also that there is not a list of unit names supplied. The nice thing here is that all of the code you wrote would be used in a single method. 

I think that by doing it this way we can have a more intuitive user experience.